app/pipeline.h

```diff
@@ -0,0 +1,47 @@
+/// \file
+/// Maintainer: Felice Serena
+///
+///
+
+#pragma once
+
+
+/// Note: This class is mostly mocked, an improved interface and implementation comes later.
+
+
+namespace MouseTrack {
+
+/// This object is responsible to coordinate the processing steps
+///
+/// - It knows how to acquire new frames
+/// - It knows how to push the frames through the pipeline steps
+/// - It informs other objects about (inbetween-)results via an observer pattern (PipelineObserver)
```

Comment:
comes with the next pull request, I'll leave it here

Reply:
What does join() do? I'm not sure it's explained clearly in the code.

Reply:
Oh, you're right, I totally forgot to mention the entire worker thread part. I'll put a note.

Reply:
Done, I hope it's now a bit clearer. We only need a worker thread, because we're working with a gui. We need one thread updating the views and interacting with the user. And we need one other to do the actual work. Doing everything in the run loop of qt could be possible, but would cause our interface to freeze terribly as soon as we do some processing.